src/app/ngrx/effects/castle.effects.ts

```diff
@@ -72,57 +72,66 @@ export class CastleEffects {
     });
 
   @Effect()
-  loadAllTimePodcastMetrics$: Observable<Action> = this.actions$
-    .ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_ALL_TIME_METRICS_LOAD)
-    .switchMap(() => {
-      if (this.selectedPodcast) {
-        return this.castle
-          .followList('prx:podcast', {id: this.selectedPodcast.feederId})
-          .map(metrics => {
-            const allTimeDownloads = metrics[0]['downloads']['total'];
-            return new ACTIONS.CastlePodcastAllTimeMetricsSuccessAction({podcast: this.selectedPodcast, allTimeDownloads});
-          })
-          .catch(error => {
-            if (error.status === 404) {
-              const allTimeDownloads = 0;
-              return Observable.of(new ACTIONS.CastlePodcastAllTimeMetricsSuccessAction({podcast: this.selectedPodcast, allTimeDownloads}));
-            } else {
-              Observable.of(new ACTIONS.CastlePodcastAllTimeMetricsFailureAction({podcast: this.selectedPodcast, error}));
-            }
-          });
-      } else {
-        // gonna be difficult to capture this error without changing shape of podcast metrics state since it's just an array
-        return Observable.of(new ACTIONS.CastlePodcastAllTimeMetricsFailureAction({podcast: undefined, error: 'No selected podcast yet'}));
-      }
+  loadPodcastPerformanceMetrics$: Observable<Action> = this.actions$
+    .ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_PERFORMANCE_METRICS_LOAD)
+    .map((action: ACTIONS.CastlePodcastPerformanceMetricsLoadAction) => action.payload)
+    .switchMap((payload: ACTIONS.CastlePodcastPerformanceMetricsLoadPayload) => {
+    const { seriesId, feederId } = payload;
+      return this.castle
+        .followList('prx:podcast', {id: feederId})
```

Comment:
I also noticed that castle and/or haldoc is misbehaving on this `prx:podcast` link ... should be a regular `follow()`, except that throws an error.  But you found something that does work, so :+4:!

Reply:
Credit to gcampo88 who set this up initially for the summary table